src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java

```diff
@@ -626,6 +629,32 @@ void testFindAllDatasetsByUser() {
     verify(datasetDAO, times(0)).findDatasetsByAuthUserEmail(any());
   }
 
+  @Test
+  void testFindAllDatasetsAsStreamingOutput() throws Exception {
+    var datasets = getDatasets(RandomUtils.nextInt(10, 20));
+    when(datasetDAO.findAllDatasetIds()).thenReturn(datasets.stream().map(Dataset::getDataSetId).toList());
+    datasets.forEach(d -> {
+      System.out.println(d);
+      when(datasetDAO.findDatasetsByIdList(List.of(d.getDataSetId()))).thenReturn(List.of(d));
+    });
+    initService();
+    // The following forces the number of calls to datasetDAO.findDatasetsByIdList to be the same as
+    // the number of datasets generated in the test.
+    datasetService.setDatasetBatchSize(1);
+
+    var output = datasetService.findAllDatasetsAsStreamingOutput();
+    var baos = new ByteArrayOutputStream();
+    output.write(baos);
+    var datasetsJson = baos.toString();
+    var listOfDatasetsType = new TypeToken<List<Dataset>>() {}.getType();
+    var gson = GsonUtil.buildGson();
+    List<Dataset> returnedDatasets = gson.fromJson(datasetsJson, listOfDatasetsType);
+    assertFalse(returnedDatasets.isEmpty());
+    assertEquals(datasets.size(), returnedDatasets.size());
+    datasets.forEach(d -> assertTrue(returnedDatasets.contains(d)));
```

Comment:
I think that the `contains` matcher can also be useful here:
```suggestion
    assertThat(returnedDatasets, contains(datasets.toArray());
```

Reply:
Or `containsInAnyOrder` if order is not guaranteed.

Reply:
In this case, we added an ID sort in the query, so I do actually care about order.